### DISCUSSION TITLE
Processing time

### DISCUSSION THREAD

**Discussion by dgutson** [oss-review-toolkit/ort#6894] [General]
I'd like to know how long ORT takes for processing your projects.
Ours takes a full night (more than 6 hours).

**Comment by sschuberth** [comment]
Can you be a bit more specific what you mean by "processing"? Which of the ORT tools do you run? For example, usually `ort analyze` is rather quick even for larger projects. IIRC I've never seen it take more than maybe 30 minutes, and that was for a massive project (using one of the package managers where we need to do suboptimal things).

On the other hand, `ort scan` usually takes long (hours to even days) when scanning a large project (in the sense of lines of code, but also in the sense of number of dependencies) *for the first time*. If you've set up a provenance-based scan storage, subsequent scans should be *much* faster. However, there are exceptions. For example if you're having a setup with a big monorepo. Then any update to that repo invalidates the cache (as the hash of the previously scanned revision changes), and everything needs to be rescanned.

You see, we'd need a bit more information about your case to be able to give a sophisticated answer.
